packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -225,6 +254,49 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
       pointerSignalResolver.resolve(event);
     }
   }
+
+  void _scheduleFrameCallback() {
+    if (_frameCallbackScheduled) {
+      return;
+    }
+    final SchedulerBinding? instance = scheduler ??= SchedulerBinding.instance;
+    assert(instance != null);
+    if (instance != null) {
+      _frameCallbackScheduled = true;
+      instance.scheduleFrameCallback((_) {
+        _frameCallbackScheduled = false;
+        _sampleTime = instance.currentSystemFrameTimeStamp + samplingOffset;
+        if (!locked) {
+          _flushPointerEventQueue();
+        }
+      });
+    }
+  }
+
+  // Flag to track if a frame callback has been scheduled.
+  bool _frameCallbackScheduled = false;
+
+  // Current sample time for resampling.
+  Duration _sampleTime = const Duration();
+
+  // Resamplers used to filter incoming touch events when resampling
+  // is enabled.
+  final Map<int, PointerEventResampler> _resamplers =
+      <int, PointerEventResampler>{};
+
+  /// Enable pointer event resampling by setting this to `true`.
+  ///
+  /// Resampling results in smoother event processing on devices
+  /// without high frequency sensors at the cost of adding some latency.
+  bool resamplingEnabled = false;
+
+  /// Offset relative to current frame time that should be used for
+  /// resampling.
+  Duration samplingOffset = _defaultSamplingOffset;
+
+  /// Scheduler binding used for frame callbacks when resampling.
+  /// Singleton instance of this object will be used when not set.
+  SchedulerBinding? scheduler;
```

Comment:
Is this only used for tests? If possible, it seems to be safer to override `SchedulerBinding.instance` in the test instead of providing one more changeable field for `GestureBinding`.

As a binding (singleton), it seems to be sufficient to just instantiate a subclass binding. For example, see https://github.com/flutter/flutter/blob/88118bcbd1e20de25875e43da62ceea7357feb4e/packages/flutter_test/lib/src/_binding_io.dart#L24-L28 and _https://github.com/flutter/flutter/blob/88118bcbd1e20de25875e43da62ceea7357feb4e/packages/flutter/test/services/binding_test.dart#L59_

Reply:
Done.